core/celery/tasks/generate_inventory_and_hold_reports.py

```diff
@@ -0,0 +1,292 @@
+        return container_instance()
+
+    def run(self) -> None:
+        with self.transaction() as session:
+            library = get_one(session, Library, id=self.library_id)
+
+            if not library:
+                self.log.error(
+                    f"Cannot generate inventory and holds report for library (id={self.library_id}): "
+                    f"library not found."
+                )
+                return
+
+            try:
+                current_time = datetime.now()
+                date_str = current_time.strftime("%Y-%m-%d_%H:%M:%s")
+                attachments: dict[str, Path] = {}
+
+                file_name_modifier = f"{library.short_name}-{date_str}"
+
+                # resolve integrations
+                integrations = session.scalars(
+                    select(IntegrationConfiguration)
+                    .join(IntegrationLibraryConfiguration)
+                    .where(
+                        IntegrationLibraryConfiguration.library_id == self.library_id,
+                        IntegrationConfiguration.goal == Goals.LICENSE_GOAL,
+                        not_(
+                            IntegrationConfiguration.settings_dict.contains(
+                                {"include_in_inventory_report": False}
+                            )
+                        ),
+                    )
+                ).all()
+                registry = LicenseProvidersRegistry()
+                integration_ids: list[int] = []
+                for integration in integrations:
+                    settings = registry[integration.protocol].settings_load(integration)
+                    if not isinstance(settings, OPDSImporterSettings):
+                        continue
+                    integration_ids.append(integration.id)
+
+                # generate inventory report csv file
+                sql_params: dict[str, Any] = {
+                    "library_id": library.id,
+                    "integration_ids": tuple(integration_ids),
+                }
+
+                inventory_report_file_path = self.generate_inventory_report(
+                    session, sql_params=sql_params
+                )
+
+                # generate holds report csv file
+                holds_report_file_path = self.generate_holds_report(
+                    session, sql_params=sql_params
+                )
+
+                attachments[f"palace-inventory-report-{file_name_modifier}.csv"] = Path(
+                    inventory_report_file_path
+                )
+                attachments[f"palace-holds-report-{file_name_modifier}.csv"] = Path(
+                    holds_report_file_path
+                )
+
+                self.services.email.send_email(
+                    subject=f"Inventory and Holds Reports {current_time}",
+                    receivers=[self.email_address],
+                    text="",
+                    attachments=attachments,
+                )
+            except Exception as e:
```

Comment:
Instead of catching the exception here, and logging a failure. I think we should drop the `except` block and just let the exception raise. Celery will log the exception and mark the job as failed. That way when we start collecting information about tasks this would count towards the failed job stat.